### PR TITLE
AccountView: Centralize user creation

### LINKED ISSUE #138: Delete a user if creation fails
### What Happened?

I set up my user and got an error that the user wasn't able to be added to the samba share. Initial setup then exited, and I was left stranded with a disabled account.

### Steps to Reproduce

No idea how to reproduce the failure, but basically failure states aren't well handled here

### Expected Behavior

We probably shouldn't exit Initial Setup here if creation fails: https://github.com/elementary/initial-setup/blob/e34c159d4e5d46b6d74a560ed59c47861c610c5a/src/MainWindow.vala#L93

And we should probably clean up any disabled/failed user accounts

### OS Version

7.x (Early Access)

### Software Version

Compiled from git

### Log Output

_No response_

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #138 

Doesn't try to address the samba share issue, but centralizes user creation instead of doing it across 3 classes and if something fails, we don't continue trying to set up and we don't quit. This should at least make sure we don't leave someone in a soft locked state

### CODE REVIEW

File: src/Views/AccountView.vala [hunk 1]
```diff
@@ -265,6 +284,137 @@ public class Installer.AccountView : AbstractInstallerView {
...
+    }
+
+    private string gen_username (string fullname) {
+        string username = "";
+        bool met_alpha = false;
+
+        foreach (char c in fullname.to_ascii ().to_utf8 ()) {
+            if (c.isalpha ()) {
+                username += c.to_string ().down ();
+                met_alpha = true;
```
Comment (zeebok) [inline src/Views/AccountView.vala:409]: I just want to note that this means that usernames cannot start with numbers. Is that by design? The warning message, at least in the user management in system settings, if you have a digit first says that the 

> username can only be lowercase letters or numbers

Improper message or bug?

**Review by zeebok — CHANGES_REQUESTED** [review]
This feels cleaner to me so I think this is a good move. Just wonder about the username generation

File: src/Views/AccountView.vala (hunk 1 above)
Comment (danrabbit) [reply src/Views/AccountView.vala:409]: So this is just for the auto-generated suggested username when you type in your real full name. I think it's probably okay to assume that a persons real full name doesn't include numbers. They can always edit the suggested username to include numbers if they like

Comment (zeebok) [reply src/Views/AccountView.vala:409]: Ah I see, I think that is a safe assumption as well. The question came around specifically with _starting_ a username with a number fails though. Dani1337 works but 2Dani2Furious will fail. Pretty minor and not really relevant to the changes this MR is for but thought I would ask.

**Review by zeebok — APPROVED** [review]

File: src/Views/AccountView.vala (hunk 1 above)
Comment (danrabbit) [reply src/Views/AccountView.vala:409]: Oh I see. Yeah we should probably handle that. Can you file an issue report?

**Pull request merged**